dev/org.eclipse.codewind.openapi.ui/src/org/eclipse/codewind/openapi/ui/commands/AbstractOpenApiGeneratorCommand.java

```diff
@@ -0,0 +1,179 @@
+	
+	protected String codegenJar = null;
+	protected String configJson = null;
+	protected String argSurround = ""; //$NON-NLS-1$
+	protected String pluginLocation = ""; //$NON-NLS-1$
+	protected boolean isWindows = true;
+	protected IProject project = null;
+
+	protected String language = null;
+	protected String generatorType = null;
+	
+	public AbstractOpenApiGeneratorCommand() {
+		// empty
+	}
+
+	public AbstractOpenApiGeneratorCommand(ISchedulingRule rule) {
+		super(rule);
+	}
+
+	public void setProject(IProject project) {
+		this.project = project;
+	}
+	
+	public void setOpenApiFile(IFile openApiFile) {
+		this.openApiFile = openApiFile;
+	}
+
+	public void setOutputFolder(IFolder outputFolder) {
+		this.outputFolder = outputFolder;
+	}
+
+	public void setLanguage(String language) {
+		this.language = language;
+	}
+	
+	public void setGeneratorType(String generatorType) {
+		this.generatorType = generatorType;
+	}
+	
+	public void setOutputFolderString(String outputFolderString) {
+		this.outputFolderString = outputFolderString;
+	}
+	
+	public String getOutputFolderString() {
+		return outputFolderString;
+	}
+
+	public IFolder getOutputFolder() {
+		return outputFolder;
+	}
+
+	
+	@Override
+	protected void execute(IProgressMonitor monitor)
+			throws CoreException, InvocationTargetException, InterruptedException {
+		try {
+			monitor.beginTask("Generating", 15);
+			List<String> mainArgsList = new ArrayList<String>();		
+			IStatus result = commonSetup(monitor);
+			monitor.worked(5);
+		    if (!result.isOK()) {
+		    	Activator.log(IStatus.ERROR, "Failed to set up command");
+		    	return;
+		    }
+		    monitor.worked(1);
+			populateArgsList(mainArgsList);
+			monitor.worked(4);
+			result = doExecute(mainArgsList, monitor);
+			monitor.worked(5);
+		} finally {
+			monitor.done();
+		}
+ 	}
+
+	protected abstract CommandRunner getCommandRunner(List<String> mainArgsList);
+	protected abstract void populateArgsList(List<String> mainArgsList);
+	
+	protected IStatus commonSetup(IProgressMonitor monitor) {
+		String osName = System.getProperty("os.name").toLowerCase(); //$NON-NLS-1$
+		isWindows = osName.indexOf("win") >= 0; //$NON-NLS-1$
+		
+		// only surround windows command line arguments with double quotes
+		if (isWindows) {
+			argSurround = "\""; //$NON-NLS-1$
+		}
+		
+		try {
+			URL coreURL = FileLocator.resolve(Activator.getDefault().getBundle().getEntry("/")); //$NON-NLS-1$
+			pluginLocation = coreURL.getFile();
+			if (isWindows && pluginLocation != null && pluginLocation.startsWith("/")) { //$NON-NLS-1$
+				pluginLocation = pluginLocation.substring(1);
+			}
+		} catch (IOException e) {
+			return new Status(IStatus.ERROR, org.eclipse.codewind.openapi.ui.Activator.PLUGIN_ID, e.getMessage());
+		}
+		
+		String jarOverride = System.getProperty(IOpenApiConstants.OPENAPI_CLI_OVERRIDE_PROPERTY);
+		// eg. -Dorg.ecliipse.codewind.openapi.cli.jar.path=/Users/kchong/openapi/openapi-generator-cli-4.0.0.jar
```

Comment:
Just confirming, do you want your OS username put on this comment Keith?

Reply:
It was intentional and served as a real-world example.  I could change it to /Users/foo/